src/main/java/com/github/coleb1911/ghost2/commands/modules/fun/ModuleFortune.java

```
@@ -0,0 +1,94 @@
+        // if an invalid arg is provided, return a help message
+        if (!ctx.getArgs().isEmpty()) {
+            categorySelected  = ctx.getArgs().get(0);
+            if (Arrays.asList(FORTUNE_CATEGORIES).contains(categorySelected)) {
+                category = categorySelected;
+            } else {
+                // This looks weird because had to apply the fencepost problem to format this correctly
+                String reply = "Invalid argument. List of available categories: " + FORTUNE_CATEGORIES[0] + " ";
+                for(int i = 1; i < FORTUNE_CATEGORIES.length - 1; i++) {
+                    reply += ", " + FORTUNE_CATEGORIES[i];
+                }
+                ctx.replyBlocking(reply);
+                return;
+            }
+        } else {
+            int index = random.nextInt(FORTUNE_CATEGORIES.length - 1);
+            category = FORTUNE_CATEGORIES[index];
+        }
+
+        // make a url and cast it as a HttpURLConnection
+        String idURL = API_URL + category;
+        URL url = new URL(idURL);
+        HttpURLConnection conn = (HttpURLConnection)url.openConnection();
+
+        conn.setRequestMethod("GET");
+        conn.connect();
+
+        int responseCode = conn.getResponseCode();
+
+        // read response
+        String json = "";
+        if(responseCode != 200) {
+            throw new RuntimeException("Response Code" + responseCode);
+        } else {
+            Scanner scanner = new Scanner(url.openStream());
+            while(scanner.hasNext()) {
+                json += scanner.nextLine();
+            }
+
+            scanner.close();
+
+            // parse JSON
+            try {
+                JSONObject obj = new JSONObject(json);
+                String fortune = (String) obj.get("fortune");
+                ctx.replyBlocking(fortune);
+            } catch (JSONException e) {
+                System.out.println("JSON parsing Exception");
```

Try not to silently fail when you can. Let the user know.

Also, sidenote: `System.out.println` is heresy when there's a logger available!!
You would replace it with TinyLog's `Logger#debug` here.

---

ahhh, heresy!
jokes aside, thanks for the heads up 😄